Documentation/observability/metrics.rst

```diff
@@ -291,6 +291,7 @@ Services
 Name                                       Labels                                             Default    Description
 ========================================== ================================================== ========== ========================================================
 ``services_events_total``                                                                     Enabled    Number of services events labeled by action type
+``service_implementation_delay``           ``action``                                         Enabled    Duration in seconds to propagate the data plane programming of a service, its network and endpoints from the time the service or the service pod was changed excluding the event queue latency
```

Comment:
> I've tested out by adding the initial start time in service_cache/emitEvent but the average was under 100ns.

Some non negligible time is usually spent here, before the event is created: https://github.com/cilium/cilium/blob/2311f3d7811c8950b7f5e1e2ac290f8216be0dc3/pkg/k8s/service_cache.go#L453

I'm okay with this PR as it is now, just FYI if that's something you want to include in metric as well.